### ISSUE TITLE
An Assert in DiagnosticBagExtensions.Add in broken solution

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#13926]
The following shows a failure in the VB compiler (preview 5, opening Roslyn.sln in Release mode) in full solution analysis mode

![image](https://cloud.githubusercontent.com/assets/3804346/18654409/c58b6af2-7e95-11e6-8905-435cfc401f0b.png)

![image](https://cloud.githubusercontent.com/assets/3804346/18654413/cc16bbba-7e95-11e6-9843-1ad856219f58.png)

this is after failing to load some generated parts of the solution.

**Comment by gafter** [comment]
The offending diagnostic was a warning that an imported namespace is empty or was not found.

**Comment by AlekseyTs** [comment]
This is just an assert.

**Comment by AlekseyTs** [comment]
I don't think this issue will cause a real crash in the compiler. However, the underlying problem manifests itself in the fact that compiler doesn't report any errors when an unresolved alias is used. This is inconsistent with behavior of the native compiler. Basically for such erroneous code, I was able to get emit to fail with a general error that doesn't point to the place where the problem was introduced in source. In other scenario, I was able to get compiler to successfully produce an invalid assembly (doesn't PEVerify and run).
